### ISSUE TITLE
Detect unnecessary usage of private property instead of field

### ISSUE THREAD

**Issue by Thieum** [meziantou/Meziantou.Analyzer#1158]
**Is your feature request related to a problem? Please describe.**
A private property declared with standard getter / setter is usually replaceable by a field, the property infrastructure doesn't add anything.

**Describe the solution you'd like**
- Detect private property with standard getter / setter, both read/write and read-only.
- Ignore properties that have special logic defined in the getter and/or the setter for the warning. 
- Ignore properties that have attributes ( for example `GeneratedRegex` )

```csharp
Dictionary<string, string> _values { get; } = new(StringComparer.OrdinalIgnoreCase);
Dictionary<string, string> _otherValues { get; set; } = new(StringComparer.OrdinalIgnoreCase);
```
- Warn and propose a fix adapted to the situation : field  or readonly field.

```csharp
readonly Dictionary<string, string> _values = new(StringComparer.OrdinalIgnoreCase);
Dictionary<string, string> _otherValues = new(StringComparer.OrdinalIgnoreCase);
```

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by meziantou** [comment]
There are more cases to handle:
- Serializations attributes (many libraries)
- partial properties
- Reflection may be broken

Which actual problem does it solve? One line is still one line. You remove some metadata from the dll, but it may not be a game changer. Performance are identical.

I'm not sure the rule is relevant and safe enough to be implemented.

**Comment by Thieum** [comment]
You're right, it's more a matter of coding style, so it might be not a valid general rule.
